edgelet/edgelet-http/src/util/incoming.rs

```diff
@@ -70,45 +71,50 @@ impl Stream for Incoming {
+                    // passed to the stream selector or has failed and needs the error bubbled up.
+                    // Return a tuple containing the index and state.
+                    let val = connections
+                        .iter_mut()
+                        .map(|(fut, _)| fut.poll())
+                        .enumerate()
+                        .find(|(_, result)| match result {
+                            Ok(v) => v.is_ready(),
+                            Err(_) => true,
+                        });
 
-                // Validate that the poll is ready, and remove that value. Then return the connection.
-                // If no connections are available in connection manager, return Async::NotReady
-                match val {
-                    Some((i, result)) => {
-                        let (_, addr) = connections.remove(i);
-                        match result {
-                            Ok(Async::Ready(tls_stream)) => {
-                                #[cfg(not(windows))]
-                                {
-                                    Async::Ready(Some((StreamSelector::Tls(tls_stream), addr)))
+                    // Validate that the poll is ready, and remove that value. Then return the connection.
+                    // If no connections are available in connection manager, return Async::NotReady
+                    match val {
+                        Some((i, result)) => {
+                            let (_, addr) = connections.remove(i);
+                            match result {
+                                Ok(Async::Ready(tls_stream)) => {
+                                    #[cfg(not(windows))]
+                                    {
+                                        return Ok(Async::Ready(Some((
+                                            StreamSelector::Tls(tls_stream),
+                                            addr,
+                                        ))));
+                                    }
+                                    #[cfg(windows)]
+                                    {
+                                        return Async::Ready(Some(
+                                            (Box::new(StreamSelector::Tls(tls_stream), addr)),
+                                        ));
+                                    }
                                 }
-                                #[cfg(windows)]
-                                {
-                                    Async::Ready(Some(
-                                        (Box::new(StreamSelector::Tls(tls_stream), addr)),
-                                    ))
+                                // The prior block included a filter that specifically asked for is_ready state,
+                                // so this line is unreachable.
+                                Ok(_) => unreachable!(),
+                                Err(err) => {
```

Comment:
Is there any error filtering that we want to do or just drop connection on any/all errors?

Reply:
It rather complicated to implement (I don't want to say impossible), but
1. This error type doesn't support ErrorKind - difficult to filter out the error
2. This is an error that comes from native_tls, so I assume that it is the type we are interested in.

During a quick discussion with @arsing we decided that it is good enough to leave like this.